### DISCUSSION TITLE
MySQL 8.4 failing to connect

### DISCUSSION THREAD

**Discussion by B4nan** [sidorares/node-mysql2#2641] [General]
MySQL 8.4 update came in via renovate a few days ago, and all the mysql tests now break as they cannot connect to the database.

https://github.com/mikro-orm/mikro-orm/actions/runs/8921763592/job/24502529378?pr=5519

Not sure if you already use MySQL 8.4 in the tests here, or if this is expected and needs some changed in how we create the MySQL 8.4 instance, so opening just a discussing for now, since I don't have time to experiment more and create an isolated repro, but wanted to let you know this problem exists.

**Comment by sidorares** [comment]
`connect ECONNREFUSED 127.0.0.1:3308`

It looks like it's not listening on `127.0.0.1:3308`. Can you double check your server config? Is it you actually configured to bind to port 3308? ( default port is 3306 ). Is it listening on 127.0.0.1? It might listen on ipv6 or not listen tcp port at all - some servers by default listen unix socket only

**Reply by B4nan** [reply]
It's a dependency upgrade, it was working fine with v8.3, it's not like I am setting this up for the first time. 

https://github.com/mikro-orm/mikro-orm/blob/master/docker-compose.yml#L14
